modules/402-ingress-nginx/hooks/minimal_ingress_nginx_version.go

```diff
@@ -80,6 +81,9 @@ func discoverMinimalNginxVersion(input *go_hook.HookInput) error {
 		ctrl := s.(ingressNginxController)
 		if ctrl.Version == "" {
```

Comment:
If the controller has explicitly specified "0.33", `isDisruptionUpdate` will not become `true`, and hence `hasDisruptionVersionUpdate` will not become `true`. 

If it is desired, please leave a comment, that we migrate only defaults, not *all* outdated controllers. 

If it is not desired, I suggest not to nest the second `if`

```go
if ctrl.Version == "" {
    ctrl.Version = input.Values.Get("ingressNginx.defaultControllerVersion").String()
}

if ctrl.Version == "0.33" {
    isDisruptionUpdate = true
}
```

Reply:
If controller is explicitly specified - it's out of this scope. User will have 5 more releases to migrate manually/controllable to the new one. We have an alert about deprecated releases. Picture looks like that:
1. In the release 1.35 we will add this check
2. In the release 1.36 we will change default controller. This will trigger disruption update
3. In the release 1.40 we will remove controllers <1.1. This action will be protected with requirement `"ingressNginx": "1.1"`